pytd/tests/test_writer.py

```diff
@@ -81,6 +81,16 @@ def test_write_dataframe_invalid_if_exists(self):
                 pd.DataFrame([[1, 2], [3, 4]]), self.table, if_exists="bar"
             )
 
+    def test_query_builder(self):
+        df = pd.DataFrame({"a": [1, 2], "b": [None, 3]})
```

Comment:
Still have an issue with [Nullable Integer type](https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html). Can you check it?

```python
In [28]: df = pd.DataFrame({'a': pd.array([1, 2, np.nan], dtype="Int64"), 'b': [4.0, None, 6.0], 'c':[None
    ...: , "none", "9"], 'd': [10, np.nan, 12.0]})

In [29]: df
Out[29]:
     a    b     c     d
0    1  4.0  None  10.0
1    2  NaN  none   NaN
2  NaN  6.0     9  12.0

In [30]: td.to_td(df, "aki.pytd_test", con, if_exists="overwrite", index=False, writer="insert_into")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-30-93a602a9721b> in <module>
----> 1 td.to_td(df, "aki.pytd_test", con, if_exists="overwrite", index=False, writer="insert_into")

~/src/pytd/pytd/pandas_td/__init__.py in to_td(frame, name, con, if_exists, time_col, time_index, index, index_label, chunksize, date_format, writer)
    394
    395     database, table = name.split(".")
--> 396     con.get_table(database, table).import_dataframe(frame, writer, mode)
    397
    398

~/src/pytd/pytd/table.py in import_dataframe(self, dataframe, writer, if_exists, **kwargs)
    115             writer = Writer.from_string(writer, **kwargs)
    116
--> 117         writer.write_dataframe(dataframe, self, if_exists)
    118
    119         if writer_from_string:

~/src/pytd/pytd/writer.py in write_dataframe(self, dataframe, table, if_exists)
    103             raise RuntimeError("this writer is already closed and no longer available")
    104
--> 105         _cast_dtypes(dataframe)
    106
    107         column_names, column_types = [], []

~/src/pytd/pytd/writer.py in _cast_dtypes(dataframe, inplace)
     41         else:
     42             t = str
---> 43         df[column] = df[column].astype(t)
     44
     45         # Bulk Import API internally handles boolean string as a boolean type,

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/generic.py in astype(self, dtype, copy, errors, **kwargs)
   5689             # else, only a single dtype is given
   5690             new_data = self._data.astype(dtype=dtype, copy=copy, errors=errors,
-> 5691                                          **kwargs)
   5692             return self._constructor(new_data).__finalize__(self)
   5693

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/internals/managers.py in astype(self, dtype, **kwargs)
    529
    530     def astype(self, dtype, **kwargs):
--> 531         return self.apply('astype', dtype=dtype, **kwargs)
    532
    533     def convert(self, **kwargs):

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/internals/managers.py in apply(self, f, axes, filter, do_integrity_check, consolidate, **kwargs)
    393                                             copy=align_copy)
    394
--> 395             applied = getattr(b, f)(**kwargs)
    396             result_blocks = _extend_blocks(applied, result_blocks)
    397

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/internals/blocks.py in astype(self, dtype, copy, errors, values, **kwargs)
    532     def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
    533         return self._astype(dtype, copy=copy, errors=errors, values=values,
--> 534                             **kwargs)
    535
    536     def _astype(self, dtype, copy=False, errors='raise', values=None,

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/internals/blocks.py in _astype(self, dtype, copy, errors, values, **kwargs)
    614
    615                 if self.is_extension:
--> 616                     values = self.values.astype(dtype)
    617                 else:
    618                     if issubclass(dtype.type,

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/arrays/integer.py in astype(self, dtype, copy)
    450         # coerce
    451         data = self._coerce_to_ndarray()
--> 452         return astype_nansafe(data, dtype, copy=None)
    453
    454     @property

~/src/pytd/.venv/lib/python3.6/site-packages/pandas/core/dtypes/cast.py in astype_nansafe(arr, dtype, copy, skipna)
    681         # work around NumPy brokenness, #1987
    682         if np.issubdtype(dtype.type, np.integer):
--> 683             return lib.astype_intsafe(arr.ravel(), dtype).reshape(arr.shape)
    684
    685         # if we have a datetime/timedelta array of objects

pandas/_libs/lib.pyx in pandas._libs.lib.astype_intsafe()

ValueError: cannot convert float NaN to integer
```

Reply:
Nice catch...how about this? c1400e8915b72710b57c918e6365543e1776fe7b

Reply:
ValueError has been fixed 👍 

Faces some type conversion mismatch for `Int64` and `boolean`, but it's OK to discuss as a separate topic since bulk_import also has similar conversion.

```python
In [1]: import os

In [2]: import numpy as np

In [3]: import pytd.pandas_td as td

In [4]: import pandas as pd

In [5]: df = pd.DataFrame({'a': pd.array([1, 2, np.nan], dtype="Int64"), 'b': [4.0, None, 6.0], 'c':[None,
   ...:  "none", "9"], 'd': [10, np.nan, 12.0], 'e': [True, False, np.nan]})

In [6]: con = td.connect(apikey=os.environ["TD_API_KEY"], endpoint=os.environ["TD_API_SERVER"])

In [7]: td.to_td(df, "aki.pytd_test", con, if_exists="overwrite", index=False, writer="insert_into")
column 'c' has non-numeric. The values are stored as 'varchar' type on Treasure Data.
column 'e' has non-numeric. The values are stored as 'varchar' type on Treasure Data.

In [8]: df2 = td.read_td_table('aki.pytd_test', con.default_engine)

In [12]: df
Out[12]:
     a    b     c     d      e
0    1  4.0  None  10.0   True
1    2  NaN  none   NaN  False
2  NaN  6.0     9  12.0    NaN

In [9]: df2
Out[9]:
     a    b     c     d      e        time
0  1.0  4.0  None  10.0   True  1564377826
1  2.0  NaN  none   NaN  False  1564377826
2  NaN  6.0     9  12.0    nan  1564377826

In [10]: df2.e
Out[10]:
0     True
1    False
2      nan
Name: e, dtype: object

In [11]: df2.e[2]
Out[11]: 'nan'
```

Reply:
@takuti Just confirmation, nullable integer for pandas doesn't work for this case?
https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html

Reply:
https://github.com/treasure-data/pytd/pull/37#issuecomment-515908945